src/Tests/dotnet.Tests/dotnet-msbuild/GivenDotnetBuildInvocation.cs

```diff
@@ -84,7 +88,33 @@ public void MsbuildInvocationIsCorrect(string[] args, string expectedAdditionalA
                     .Should()
                     .Be($"{ExpectedPrefix} -nologo -consoleloggerparameters:Summary{expectedAdditionalArgs}");
             });
+        }
 
+        [Fact]
+        public void OsOptionIsCorrectlyResolved()
+        {
+            CommandDirectoryContext.PerformActionWithBasePath(WorkingDirectory, () =>
+            {
+                var msbuildPath = "<msbuildpath>";
+                var command = BuildCommand.FromArgs(new string[] { "--os", "os" }, msbuildPath);
+                var expectedArch = Environment.Is64BitOperatingSystem ? "x64" : "x86";
```

Comment:
What if the OS is arm64?

Reply:
This is just a test, so it's not accounting for that right now.

Reply:
But is should, right? I know the SDK repo doesn't run tests on arm64, and unfortunately the installer doesn't do that either right now. But we're trying to enable the tests on installer (at least I would hope so, we can't ship a supported platform without ever running SDK tests on it...), once that happens, this will likely break.

Reply:
Updated in latest commit